src/serve/proxy/ngrok-proxy.ts

```diff
@@ -36,7 +36,10 @@ export class NgrokProxy implements ProxyInterface {
 
     listener.forward(`${targetHost}:${targetPort}`);
 
-    return new URL(url);
+    const urlWithPath = new URL(url);
+    urlWithPath.pathname = targetUrl.pathname;
+    urlWithPath.search = targetUrl.search;
```

Comment:
It's uncommon for LIFF app endpoint URLs to include a hash, but could you also include the hash in `urlWithPath`, just like `pathname` and `search`, to ensure that `NgrokProxy` correctly forwards the entire URL?